### PR TITLE
Prop tests

### PR DESCRIPTION
**Pull request by lepsa**
Some simple property tests for a handful of maths operators.

### CODE REVIEW

File: tests/PropertySpec/Exec.hs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  )where
+
+import           Data.Foldable                               (fold, foldl1)
+import           Data.List.NonEmpty                          (intersperse)
+import           ExecSpec.Util                               (num)
+import           Graphics.Implicit.ExtOpenScad.Definitions   (OVal (ONum))
+import           Graphics.Implicit.ExtOpenScad.Eval.Constant (runExpr)
+import           HaskellWorks.Hspec.Hedgehog                 (requireProperty)
+import           Hedgehog                                    (diff, forAll)
+import qualified Hedgehog.Gen                                as Gen
```
Comment (julialongtin) [inline tests/PropertySpec/Exec.hs]: still need to fully qualify imports.

File: tests/PropertySpec/Exec.hs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import qualified Hedgehog.Range                              as Range
```
Comment (julialongtin) [inline tests/PropertySpec/Exec.hs]: here too.

File: tests/PropertySpec/Exec.hs [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+import           Prelude                                     (Bool, Floating, String, Double,
+                                                              Ord, fail, show,
+                                                              ($), (&&), (+),
+                                                              (.), (<$>), (<=), (*), (/),
+                                                              (<>), (>=), (-))
+import           Test.Hspec                                  (Spec, it)
+
+approx :: (Floating a, Ord a) => a -> a -> a -> Bool
```
Comment (julialongtin) [inline tests/PropertySpec/Exec.hs]: this function and the next one deserve a one-liner comment each.

File: tests/PropertySpec/Exec.hs [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+approx z a b = a + z >= b && a <= b + z
+
+mathsSpec :: (Double -> Double -> Double) -> String -> Spec
```
Comment (julialongtin) [inline tests/PropertySpec/Exec.hs]:
```suggestion
-- FIXME: magic numbers. 100, 1000, 0.000001...
mathsSpec :: (Double -> Double -> Double) -> String -> Spec
```

File: tests/PropertySpec/Exec.hs (hunk 1 above)
Comment (lepsa) [reply tests/PropertySpec/Exec.hs]: I don't think that it is an issue, as you always know where the qualified function call comes from up to overloading the qualified namespace.

Comment (julialongtin) [reply tests/PropertySpec/Exec.hs]: Sure, but it's harder to see what in the import is being used, without digging through the whole file. I find it useful when dealing with larger libraries to look through their import lists for functions in a similar vein, and standardize on using less functions.

File: tests/PropertySpec/Exec.hs (hunk 3 above)
Comment (lepsa) [reply tests/PropertySpec/Exec.hs]: I don't think so. The function isn't exported and this pattern of comparisons is prevalent if you are comparing floating point numbers.

File: tests/PropertySpec/Exec.hs [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+opName Add = "addition"
+opName Sub = "subttraction"
+opName Mul = "multiplication"
+opName Div = "division"
+
+fromOp :: Op -> Double -> Double -> Double
+fromOp Add = (+)
+fromOp Sub = (-)
+fromOp Mul = (*)
+fromOp Div = (/)
```
Comment (lepsa) [inline tests/PropertySpec/Exec.hs]: Sum type to tie all of the names and functions together, so things aren't forgotten.

**Pull request merged**